test/test_gauged.py

```diff
@@ -549,6 +549,86 @@ def test_clear_from(self):
+            writer.add({'foo': 6, 'bar': 6}, timestamp=60000)
+            writer.add({'foo': 6, 'bar': 6}, timestamp=60000, namespace=1)
+        with gauged.writer as writer:
+            writer.clear_key_before('foo', namespace=1, timestamp=60000)
+        # 'foo' in namespace 0 is untouched before 60000
+        self.assertEqual(gauged.value('foo', timestamp=50000), 5)
+        # 'foo' in namespace 1 is correctly cleared
+        self.assertEqual(gauged.value('foo', timestamp=50000, namespace=1), None)
+        # 'bar' is still there
+        self.assertEqual(gauged.value('bar', timestamp=50000), 5)
+        self.assertEqual(gauged.value('bar', timestamp=50000, namespace=1), 5)
+
+    def test_clear_key_after(self):
+        gauged = Gauged(self.driver, resolution=1000, block_size=10000)
+        with gauged.writer as writer:
+            writer.add({'foo': 1, 'bar': 1}, timestamp=10000)
+            writer.add({'foo': 2, 'bar': 2}, timestamp=20000)
+            writer.add({'foo': 3, 'bar': 3}, timestamp=30000)
+            writer.add({'foo': 4, 'bar': 4}, timestamp=40000, namespace=1)
+        self.assertEqual(gauged.value('foo', timestamp=40000), 3)
+        self.assertEqual(gauged.value('bar', timestamp=40000), 3)
+        self.assertEqual(gauged.value('foo', timestamp=40000, namespace=1), 4)
+        self.assertEqual(gauged.value('bar', timestamp=40000, namespace=1), 4)
+        with gauged.writer as writer:
+            writer.clear_key_after('foo', timestamp=20000)
+        # every value before 20000 stays the same
+        self.assertEqual(gauged.value('foo', timestamp=10000), 1)
+        self.assertEqual(gauged.value('bar', timestamp=10000), 1)
+        # 'foo' value on 20000 is cleared
+        self.assertEqual(gauged.aggregate('foo', Gauged.COUNT, start=19000, end=21000), 0)
+        # 'foo' value after 20000 is cleared
+        self.assertEqual(gauged.aggregate('foo', Gauged.COUNT, start=20000, end=40000), 0)
+        # 'bar' stays there
+        self.assertEqual(gauged.value('bar', timestamp=10000), 1)
+
+        with gauged.writer as writer:
+            writer.add({'foo': 5, 'bar': 5}, timestamp=50000)
+            writer.add({'foo': 5, 'bar': 5}, timestamp=50000, namespace=1)
+            writer.add({'foo': 6, 'bar': 6}, timestamp=60000)
+            writer.add({'foo': 6, 'bar': 6}, timestamp=60000, namespace=1)
+        with gauged.writer as writer:
+            writer.clear_key_after('foo', namespace=1, timestamp=50000)
+        # 'foo' in namespace 0 is untouched after 50000
+        self.assertEqual(gauged.value('foo', timestamp=60000), 6)
+        # 'foo' in namespace 1 is correctly cleared
+        self.assertEqual(gauged.aggregate('foo', Gauged.COUNT, start=50000, end=60000, namespace=1), 0)
+        # 'bar' is still there
+        self.assertEqual(gauged.value('bar', timestamp=60000), 6)
+        self.assertEqual(gauged.value('bar', timestamp=60000, namespace=1), 6)
```

Comment:
Can you add a test case to both which shows the behavior when you try and delete from a timestamp that isn't on a block boundary?

Something like

```python

with gauged.writer as writer:
    with self.assertRaises(ValueError):
        writer.clear_key_after('foo', timestamp=15000)
```